test-suite/defaultprobabilitycurves.cpp

```diff
@@ -376,16 +374,10 @@ void DefaultProbabilityCurveTest::testSingleInstrumentBootstrap() {
 void DefaultProbabilityCurveTest::testUpfrontBootstrap() {
     BOOST_TEST_MESSAGE("Testing bootstrap on upfront quotes...");
 
-    SavedSettings backup;
     // not taken into account, this would prevent the upfront from being used
     Settings::instance().includeTodaysCashFlows() = false;
 
     testBootstrapFromUpfront<HazardRate,BackwardFlat>();
-
-    // also ensure that we didn't override the flag permanently
-    ext::optional<bool> flag = Settings::instance().includeTodaysCashFlows();
-    if (flag != false)
-        BOOST_ERROR("Cash-flow settings improperly modified");
```

Comment:
I think this is unrelated — it doesn't check that the test code resets the flag, but that the [code in the upfront helper](https://github.com/lballabio/QuantLib/blob/master/ql/termstructures/credit/defaultprobabilityhelpers.cpp#L296-L301) does.  We probably had some kind of regression, hence the test.  I'd keep the check.